tools/src/scie_pants/pants_version.py

```diff
@@ -148,9 +148,16 @@ def determine_latest_stable_version(
     ptex: Ptex, pants_config: Path, find_links_dir: Path, github_api_bearer_token: str | None = None
 ) -> tuple[Callable[[], None], ResolveInfo]:
     info(f"Fetching latest stable Pants version since none is configured")
-    pants_version = ptex.fetch_json("https://pypi.org/pypi/pantsbuild.pants/json")["info"][
-        "version"
-    ]
+
+    latest_tag = ptex.fetch_json(
+        "https://github.com/pantsbuild/pants/releases/latest", Accept="application/json"
+    )["tag_name"]
```

Comment:
Come to think of it, could we avoid assuming the response will always include the `tag_name`, even if that is the case today.

The day that is not the case, the error is going to be _ugly_, and we can easily do a `.get("tag_name")` instead and check that `latest_tag != None` with a proper error message.

Reply:
Yes. It uses `json` to parse.

And I'd be _really_ surprised if GitHub changed their response schema like that. They seem to be very good REST citizens. But I'm also happy to tweak.

Reply:
agree it's overly defensive, but also easy enough to do, and I prefer to protect against changes we have no control over.